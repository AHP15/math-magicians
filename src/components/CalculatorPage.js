import Calculator from './Calculator';
import NavBar from './NavBar';
import buttonsInfo from '../buttons';

const CalculatorPage = () => (
  <>
    <NavBar />
    <div style={{
      display: 'flex',
      justifyContent: 'space-between',
      padding: '100px',
    }}
    >
      <h2>You want to do some math?</h2>
      <Calculator buttons={buttonsInfo} />
    </div>
  </>
);

export default CalculatorPage;
