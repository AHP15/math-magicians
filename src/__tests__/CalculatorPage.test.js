import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import CalculatorPage from '../components/CalculatorPage';

it('renders CalculatorPage correctly', () => {
  const tree = renderer
    .create(<BrowserRouter><CalculatorPage /></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
