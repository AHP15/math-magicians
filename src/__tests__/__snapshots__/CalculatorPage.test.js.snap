// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CalculatorPage correctly 1`] = `
Array [
  <nav
    className="navbar"
  >
    <a
      className="link"
      href="/"
      onClick={[Function]}
    >
      <h1>
        Math Magicians
      </h1>
    </a>
    <ul
      className="links"
    >
      <li>
        <a
          className="link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="link"
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          className="link"
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>,
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
        "padding": "100px",
      }
    }
  >
    <h2>
      You want to do some math?
    </h2>
    <div
      className="calculator"
    >
      <div
        className="result"
        title="result"
      >
        <p>
          0
        </p>
      </div>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "auto",
          }
        }
        type="button"
      >
        AC
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "auto",
          }
        }
        type="button"
      >
        +/-
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "auto",
          }
        }
        type="button"
      >
        %
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
          }
        }
        type="button"
      >
        รท
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "auto",
          }
        }
        type="button"
      >
        7
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "auto",
          }
        }
        type="button"
      >
        8
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "auto",
          }
        }
        type="button"
      >
        9
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
          }
        }
        type="button"
      >
        x
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "auto",
          }
        }
        type="button"
      >
        4
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "auto",
          }
        }
        type="button"
      >
        5
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "auto",
          }
        }
        type="button"
      >
        6
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
          }
        }
        type="button"
      >
        -
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "auto",
          }
        }
        type="button"
      >
        1
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "auto",
          }
        }
        type="button"
      >
        2
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "auto",
          }
        }
        type="button"
      >
        3
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
          }
        }
        type="button"
      >
        +
      </button>
      <button
        className="zero"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "auto",
          }
        }
        type="button"
      >
        0
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "auto",
          }
        }
        type="button"
      >
        .
      </button>
      <button
        className=""
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
          }
        }
        type="button"
      >
        =
      </button>
    </div>
  </div>,
]
`;
